crates/uv-workspace/src/pyproject.rs

```diff
@@ -70,18 +70,43 @@ impl PyProjectToml {
     /// non-package ("virtual") project.
     pub fn is_package(&self) -> bool {
         // If `tool.uv.package` is set, defer to that explicit setting.
-        if let Some(is_package) = self
-            .tool
-            .as_ref()
-            .and_then(|tool| tool.uv.as_ref())
-            .and_then(|uv| uv.package)
-        {
+        if let Some(is_package) = self.uv_package() {
             return is_package;
         }
 
         // Otherwise, a project is assumed to be a package if `build-system` is present.
         self.build_system.is_some()
     }
+
+    /// Returns whether the project should be considered a package with scripts.
+    ///
+    /// This detects if a project is declared as a uv package, or if it has
+    /// a scripts-related section defined.
+    pub fn is_packaged_script(&self) -> bool {
```

Comment:
I'd be surprised if we needed this exact combination again. I'd just add a function like `has_scripts` and `&&` that with `is_package`.